File: API/DataLayer/Entities/Learning/LearningTask.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace API.DataLayer.Entities.Learning
{
    public class LearningTask
    {
        [Required]
        public string Tag { get; set; }
        public int LearningTaskId { get; set; }
        public int StudentId { get; set; }
        public DateTime DeadlineDate { get; set; }
        public int Importance { get; set; }

        public virtual ICollection<LectureLearningTask> LectureLearningTasks { get; set; }
    }
}

File: learn-app-client/src/app/_models/learning-task.ts
export interface LearningTask extends LearningTaskMin {
    lectures: any [];
}

export interface LearningTaskMin {
    tag: string;
    learningTaskId: number;
    studentId: number;
    importance: number;
    completed: boolean;
    deadlineDate: Date;
    completedLecturesCount: number;
    lecturesCount: number; 
}